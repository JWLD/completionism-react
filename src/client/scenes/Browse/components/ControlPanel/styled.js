import styled from 'styled-components'
import FaAlignJustify from 'react-icons/lib/fa/align-justify'
import FaTh from 'react-icons/lib/fa/th'

import { mixins, vars } from 'style'

export const SettingRow = styled.div`
  ${mixins.flexLeft};

  height: ${vars.itemHeight}rem;
  min-height: ${vars.itemHeight}rem;
  width: 100%;
  border-radius: 0.4rem;
  background-color: ${vars.barGrey};
`

export const SettingTitle = styled.span`
  margin-left: 1.5rem;
  margin-right: auto;
  color: white;
  font: 2.5rem DinMC;
`

const SettingIcon = styled.i`
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  padding: 0.75rem;
  padding-top: 0.9rem;
  color: white;

  :hover {
    cursor: pointer;
  }
`

export const ListIcon = styled(SettingIcon.withComponent(FaAlignJustify))`
  color: ${props => (props.active ? 'white' : 'grey')};
`

export const SquaresIcon = styled(SettingIcon.withComponent(FaTh))`
  color: ${props => (props.active ? 'white' : 'grey')};
`
