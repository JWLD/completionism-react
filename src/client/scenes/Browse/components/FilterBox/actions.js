import * as ACTIONS from 'constants/action_types'

// ACTION CREATORS

export const setFilter = value => ({
  type: ACTIONS.SET_FILTER_VALUE,
  value
})

// THUNKS

export const resetFilter = () => dispatch => {
  dispatch(setFilter(''))
}
