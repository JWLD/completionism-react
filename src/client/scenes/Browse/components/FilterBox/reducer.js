import * as ACTIONS from 'constants/action_types'

export const initialState = {
  value: ''
}

const filterBoxReducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.SET_FILTER_VALUE:
      return { ...state, value: action.value }
    default:
      return state
  }
}

export default filterBoxReducer
