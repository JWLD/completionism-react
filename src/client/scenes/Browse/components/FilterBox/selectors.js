export const filterValueSelector = state => state.browse.filterBox.value
