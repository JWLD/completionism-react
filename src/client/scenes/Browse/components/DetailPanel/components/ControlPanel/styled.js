import styled from 'styled-components'
import FaCheck from 'react-icons/lib/fa/check'

import { mixins, vars } from 'style'

const settingPadding = vars.itemListPadding * 0.75
const checkBoxHeight = 2.2

export const Setting = styled.div`
  ${mixins.flexVertical};

  width: 100%;
  padding: ${settingPadding}rem;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
`

export const SettingTitle = styled.span`
  margin: -0.3rem 0 ${settingPadding * 0.5}rem 0;
  font: 2.25rem DinMC;
  text-transform: uppercase;
`

export const Option = styled.button`
  ${mixins.flex};

  justify-content: space-between;
  width: 100%;
  background-color: transparent;
  color: white;
  font: 2.2rem DinRC;
  line-height: 0;

  :not(:last-child) {
    margin-bottom: ${settingPadding * 0.75}rem;
  }
`

export const CheckBox = styled(FaCheck)`
  height: ${checkBoxHeight}rem;
  width: ${checkBoxHeight}rem;
  padding: 0.2rem;
  border-radius: 0.2rem;
  background-color: ${props => (props.active ? vars.pos : vars.mainGrey)};
  color: ${props => (props.active ? 'white' : vars.mainGrey)};
`
