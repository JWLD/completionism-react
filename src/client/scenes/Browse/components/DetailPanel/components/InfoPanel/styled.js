import styled from 'styled-components'
import FaRocket from 'react-icons/lib/fa/rocket'

import { mixins, vars } from 'style'

export const InfoPanel = styled.div`
  ${mixins.flexVerticalTop};

  width: 100%;
  flex-grow: 1;
  overflow: scroll;

  > *:not(:last-child) {
    margin-bottom: ${vars.itemListPadding}rem;
  }
`

export const ItemName = styled.span`
  ${mixins.flex};

  width: 100%;
  color: white;
  font: 2.5rem DinRC;
  white-space: nowrap;
`

export const ItemImage = styled.div.attrs({
  style: ({ imageUrl }) => ({
    backgroundImage: `url(${imageUrl})`
  })
})`
  width: 100%;
  padding-bottom: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  background-size: cover;
  background-position: center;
`

export const WowheadLink = styled.a`
  ${mixins.flexLeft};

  justify-content: space-between;
  width: 100%;
  height: ${vars.itemHeight}rem;
  padding: 1.2rem;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
  font: 2.5rem DinRC;

  :hover {
    background-color: #555;
  }
`

export const WowheadIcon = styled(FaRocket)`
  height: 100%;
`
