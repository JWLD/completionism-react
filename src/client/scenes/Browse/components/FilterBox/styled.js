import styled from 'styled-components'
import { FaTimesCircle } from 'react-icons/lib/fa'

import { mixins, vars } from 'style'

export const FilterInput = styled.input`
  ${mixins.browseItem};

  padding: 1rem;
  background-color: #eee;
  font: 2.5rem DinMC;
`

export const ClearInputBtn = styled(FaTimesCircle)`
  display: ${props => !props.active && 'none'};
  position: absolute;
  right: 0;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  margin-right: ${vars.itemListPadding}rem;
  padding: 0.75rem;
  color: gray;
  background-color: transparent;
  font-size: 3.25rem;

  :hover {
    cursor: pointer;
  }
`
