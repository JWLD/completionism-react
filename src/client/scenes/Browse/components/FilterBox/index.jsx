import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { resetFilter, setFilter } from 'FilterBox/actions'

import * as SC from 'FilterBox/styled'

export const FilterBox = props => (
  <SC.FilterWrap>
    <SC.FilterInput
      onChange={e => props.setFilter(e.target.value)}
      value={props.filterValue}
      placeholder="Filter"
    />
    <SC.ClearInputBtn
      active={props.filterValue}
      onClick={() => props.resetFilter()}
    />
  </SC.FilterWrap>
)

FilterBox.propTypes = {
  filterValue: PropTypes.string.isRequired,
  resetFilter: PropTypes.func.isRequired,
  setFilter: PropTypes.func.isRequired
}

export const mapStateToProps = state => ({
  filterValue: state.browse.filter
})

const mapDispatchToProps = {
  resetFilter,
  setFilter
}

export default connect(mapStateToProps, mapDispatchToProps)(FilterBox)
