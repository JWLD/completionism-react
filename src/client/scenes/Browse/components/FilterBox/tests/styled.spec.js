import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { ClearInputBtn } from 'scenes/Browse/components/FilterBox/styled'

Enzyme.configure({ adapter: new Adapter() })

let mockProps = {}

beforeEach(() => {
  mockProps = {
    active: true
  }
})

describe('<ClearInputBtn />', () => {
  it('handles [active] prop correctly when true', () => {
    const component = shallow(<ClearInputBtn {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('handles [active] prop correctly when false', () => {
    mockProps.active = false
    const component = shallow(<ClearInputBtn {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
