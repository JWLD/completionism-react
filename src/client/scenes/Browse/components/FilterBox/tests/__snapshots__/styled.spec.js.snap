// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClearInputBtn /> handles [active] prop correctly when false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClearInputBtn
    active={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "className": "sc-bZQynM kuTCkv",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "className": "sc-bZQynM kuTCkv",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ClearInputBtn /> handles [active] prop correctly when true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClearInputBtn
    active={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "className": "sc-bZQynM gdpUXn",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "className": "sc-bZQynM gdpUXn",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
